src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+**TL;DR:** here's a diagram to help you debug your deployments in Kubernetes.
+
+![Flow chart to debug deployments in Kubernetes](diagram.png)
+
+When you wish to deploy an application in Kubernetes, you usually define three components:
+
+- a Deployment — which is a recipe for creating copies of your application called Pods
+- a Service — an internal load balancer that routes the traffic to Pods
+- an Ingress — a description of how the traffic should flow from outside the cluster to your Service.
+
+Here's a quick visual recap.
+
+```slideshow
+{
+  "description": "Kubernetes core concepts: Ingress, Service, Pod and Deployment.",
+  "slides": [
+    {
+      "image": "core-concepts-1.svg",
+      "description": "In Kubernetes your applications are exposed through two layers of load balancers: internal and external."
+    },
+    {
+      "image": "core-concepts-2.svg",
+      "description": "The internal load balancer is called Service, whereas the external one is called Ingress."
+    },
+    {
+      "image": "core-concepts-3.svg",
+      "description": "Pods are not deployed directly. Instead, the Deployment creates the Pods and whatches over them."
+    }
+  ]
+}
+```
+
+Assuming you wish to deploy a simple _Hello World_ application, the YAML for such application should look similar to this:
+
+```yaml|highlight=2,23,34|title=hello-world.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: app
+  labels:
+    track: canary
+spec:
+  selector:
+    matchLabels:
+      name: app
+  template:
+    metadata:
+      labels:
+        name: app
+    spec:
+      containers:
+      - name: app
+        image: learnk8s/app:1.0.0
+        ports:
+        - containerPort: 8080
```

Comment:
idea, but ignore me if don't agree: add resource request specification, just something simple. memory 200M, cpu 0.1.

No need for limit, just request.

Without resource specification the whole kubernetes thingie is meaningless, how do you play tetris if you don't know the items you are currently putting down.

Reply:
Correct, but I could say the same for Liveness/Readiness. I try to keep the YAML as short as I can because it's already too long.

Reply:
I would also rather use [minimal working examples](https://en.wikipedia.org/wiki/Minimal_working_example) as it's less confusing for beginners.